docs/DeveloperGuide.md

```diff
@@ -1079,6 +1079,77 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+* Handling duplicates of phone, email and link across different contacts.
+  * Current behaviour: When the user attempts to add the same phone, email or link as another contact's phone, email or link respectively, the software adds the detail without any warning.
+  * Enhanced behaviour: When the user attempts to add the same phone, email or link as nother contact's phone, email or link respectively, the software should still add the detail, but with a warning message. For example, suppose that the contact at index 2 has a phone number of `12345678`, the command `add 1 /phone 12345678` gives a warning message: `Added phone "12345678" to the contact at index 1. Phone number "12345678" already exists in the contact at index 2. If you added the phone number by accident, type "undo" to undo adding the phone number.`, and the command `create /name Nguyen /phone 12345678` gives a warning message: `Noted, created new contact: Nguyen; Phones: [12345678]. Phone number "12345678" already exists in the contact at index 2. If you added the phone number by accident, type "undo" to undo adding the phone number.`
+  * Justification: Generally, phone numbers, emails and links are not shared by multiple contacts, hence adding the warning message warns the user of a possible mistake. However, the software should not prevent the user totally from doing so to accommodate the rare case that a phone number, email or link is shared by multiple contacts.
+
+* More user-friendly info message for `create` command.
+  * Current behaviour: Upon a successful `create` command, the info message is rather verbose. For example, with command `create /name Nguyen /phone 12345678 /phone 87654321`, the info message is: `Noted, created new contact: Nguyen; Phones: [12345678, 87654321]; Emails: []; Links: []; Courses: []; Specialisations: []; Tags: `.
+  * Enhanced behaviour: The info message can be shortened to show only information being added. For example, with command `create /name Nguyen /phone 12345678 /phone`, the info message can be: `Noted, created new contact: Nguyen; Phones: [12345678, 87654321].`
+
+* More user-friendly info message for `add` command.
+  * Current behaviour: Upon a successful `add` command, the info message is rather verbose. For example, with command `add 1 /phone 12345678`, (depending on your contact list) the info message might be: `Added information to this contact: Bernice Yu; Phones: [99272758, 12345678]; Emails: [berniceyu@example.com]; Links: [github.com/bernfish]; Graduation: AY2020/2021 Semester 1; Courses: [Computer Science]; Specialisations [Artificial Intelligence]; Tags: [colleagues][friends]; Priority: High`.
+  * Enhanced behaviour: The info message can be shortened to show only the information being added. For example, with command `add 1 /phone 12345678`, the info message can be: `Added phone "12345678" to the contact at index 1.`
```

Comment:
Maybe from the user's perspective it would also be helpful to show the user the contact who has been changed (e.g. Bernice Yu at index 1)